kong/llm/drivers/shared.lua

```diff
@@ -192,41 +210,62 @@ end
 -- as if it were an SSE message.
 --
 -- @param {string} frame input string to format into SSE events
--- @param {string} delimiter delimeter (can be complex string) to split by
+-- @param {boolean} raw_json sets application/json byte-parser mode
 -- @return {table} n number of split SSE messages, or empty table
-function _M.frame_to_events(frame)
+function _M.frame_to_events(frame, raw_json_mode)
   local events = {}
 
-  -- todo check if it's raw json and
-  -- just return the split up data frame
-  if string.sub(str_ltrim(frame), 1, 1) == "{" then
-    for event in frame:gmatch("[^\r\n]+") do
-      events[#events + 1] = {
-        data = event,
-      }
+  if (not frame) or (#frame < 1) or (type(frame)) ~= "string" then
+    return
+  end
+
+  -- some new LLMs return the JSON object-by-object,
+  -- because that totally makes sense to parse?!
+  if raw_json_mode then
```

Comment:
we need some unit tests for this `if`

Reply:
I've added test suite in 5a6daca3bf7d7479af6fac41635b1849349af672 now